### PR TITLE
allow oil cracker middle input hatch accept more fluids

### PR DESCRIPTION
**Pull request by GlodBlock**
to make these recipes viable.
image
image
image

### CODE REVIEW

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_OilCracker.java [hunk 1]
```diff
@@ -334,7 +335,7 @@ private void replaceDeprecatedCoils(IGregTechTileEntity aBaseMetaTileEntity) {
             tHatch.mRecipeMap = getRecipeMap();
             if (isValidMetaTileEntity(tHatch) && tHatch.getFillableStack() != null) {
                 FluidStack tStack = tHatch.getFillableStack();
-                if (tStack.isFluidEqual(GT_ModHandler.getSteam(1000)) || tStack.isFluidEqual(Materials.Hydrogen.getGas(1000))) {
+                if (tStack.isFluidEqual(GT_ModHandler.getSteam(1000)) || tStack.isFluidEqual(Materials.Hydrogen.getGas(1000)) || tStack.isFluidEqual(FluidRegistry.getFluidStack("molten.atomic separation catalyst", 1000)) || tStack.isFluidEqual(FluidRegistry.getFluidStack("molten.atomic separation catalyst", 1000)) || tStack.isFluidEqual(FluidRegistry.getFluidStack("nitricacid", 1000)) || tStack.isFluidEqual(FluidRegistry.getFluidStack("nitricacid", 1000)) || tStack.isFluidEqual(FluidRegistry.getFluidStack("plasma.silver", 1000)) || tStack.isFluidEqual(FluidRegistry.getFluidStack("methanol", 1000))) {
```
Comment (Glease) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_OilCracker.java:338]: The list is getting a bit too long. I'd rather prefer a better solution than a super duper long line.

BTW please avoid using space in fluid names. use underscores instead. There is no hard limit, but it's by convention to use underscores instead.

Comment (GlodBlock) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_OilCracker.java:338]: these space are auto generated by GT/BW material system. there's nothing I can do about it.
maybe i can make a config for the fluid list.

Comment (Glease) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_OilCracker.java:338]: We can probably add a hook into the recipe adder system.

**Review by botn365 — CHANGES_REQUESTED** [review]
1. just lock hydrogen and steam out of the side input bus sinse imo it doesent realy matter for the other recipes

2. you could be smart with the recipe and check if that fluid is the first or second fluid but that would happen evry time you try to put in fluid so would be prety lagy

3. you could also allow all fluids in evry input bus and have a modifyed recipe check this would be a better option sinse it would be way more tps frendly

4. you could do a global hash map that stores all primary fluids and a second for all secondery fluids and modify the addRecipe code to put it in there 

this would be more tps frendly then the second option but still feels prety bad

but this here NO its horible

**Comment by GlodBlock** [comment]
i make another recipe find method for the cracker.
use the order of recipe input fluid to determine whether it should be input at side or the middle. it won't be slower than the origin one, because it only search in `sCrakingRecipes` instead of the whole recipe collection.

**Comment by Dream-Master** [comment]
@Glease @botn365

**Review by Glease — APPROVED** [review]

**Comment by Dream-Master** [comment]
@botn365

**Comment by DualBallers** [comment]
image

image

Spelling here?

**Pull request merged**